haystack/pipelines/ray.py

```diff
@@ -182,16 +190,29 @@ def load_from_yaml(  # type: ignore
         )
 
     @classmethod
-    def _create_ray_deployment(cls, component_name: str, pipeline_config: dict, replicas: int = 1):
+    def _create_ray_deployment(
+        cls,
+        component_name: str,
+        pipeline_config: dict,
+        replicas: int = 1,
+        deployment_kwargs: Optional[Dict[str, Any]] = {},
+    ):
         """
         Create a Ray Deployment for the Component.
 
         :param component_name: Class name of the Haystack Component.
         :param pipeline_config: The Pipeline config YAML parsed as a dict.
         :param replicas: By default, a single replica of the component is created. It can be
                          configured by setting `replicas` parameter in the Pipeline YAML.
+        :param deployment_kwargs: An optional dictionary of arguments to be supplied to the
+                                  `ray.serve.deployment()` method, like `ray_actor_options`,
+                                  `max_concurrent_queries`, etc. See potential values in the
+                                  Ray Serve API docs (https://docs.ray.io/en/latest/serve/package-ref.html)
+                                  under the `ray.serve.deployment()` method
         """
-        RayDeployment = serve.deployment(_RayDeploymentWrapper, name=component_name, num_replicas=replicas)  # type: ignore
+        RayDeployment = serve.deployment(
+            _RayDeploymentWrapper, name=component_name, num_replicas=replicas, **deployment_kwargs  # type: ignore
+        )
```

Comment:
Does it make sense to take `num_replicas` from the `deployment_kwargs` as well?
This is a breaking change, but would make more sense to me.

Reply:
@ZanSara

Reply:
Hi @dmigo,
I agree that it would make more sense to take the `num_replicas` from the `deployment_kwargs` as well, I just did not want to introduce breaking changes. But if you and @ZanSara are okey with that, then I can add it.

Reply:
As an alternative we could simply unpack all arguments in `deployment_kwargs` and put them at the same level as `replicas`. That would prevent the breaking change.

In any case I don't think a breaking change here is an issue. Feel free to move `replicas` into `deployment_kwargs` as well if it looks better to you. In that case just mind to add the breaking-change label so we won't miss this in the release notes :slightly_smiling_face:

Reply:
I have made the `replicas` change by moving it below the new `serve_deployment_kwargs`.